### PR TITLE
Np 46966 Single object loading, wire ViewCompiler into transformation

### PR DESCRIPTION
**Pull request by monaullah**
Wire `ViewCompiler` into transformation in `SingleObjectDataLoader`.

Next Step: Wire `ViewCompiler` into transformation in bulk-load.

### CODE REVIEW

File: data-report-commons/src/main/java/commons/ViewCompiler.java [hunk 1]
```diff
@@ -21,17 +23,30 @@ public ViewCompiler(InputStream inputStream) {
...
+    private Model extractPublicationView() {
+        var query = IoUtils.stringFromResources(NVA_QUERY);
         try (var queryExecution = QueryExecutionFactory.create(query, model)) {
             return queryExecution.execConstruct();
         }
     }
+
+    private boolean isPublication(Model model) {
+        var publicationType = model.createResource(PUBLICATION);
+        return model.listResourcesWithProperty(RDF.type, publicationType).hasNext();
```
Comment (monaullah) [inline data-report-commons/src/main/java/commons/ViewCompiler.java:50]: This method is a bit sketchy, better idea @brinxmat?

**Review by brinxmat — APPROVED** [review]
See the comment

File: data-report-commons/src/main/java/commons/ViewCompiler.java (hunk 1 above)
Comment (monaullah) [reply data-report-commons/src/main/java/commons/ViewCompiler.java]: See latest commit: 
[Use id to verify resource type](https://github.com/BIBSYSDEV/nva-data-report-api/pull/146/commits/9514c156c2306b56183af650e0d458f75f5a626e)
I think this is much better

File: data-report-commons/src/main/java/commons/ViewCompiler.java [hunk 2]
```diff
@@ -35,6 +35,12 @@ public Model extractView(URI id) {
         }
     }
 
+    private static boolean isNotObject(Model model, URI id) {
+        var publication = model.createResource(id.toString());
```
Comment (brinxmat) [inline data-report-commons/src/main/java/commons/ViewCompiler.java:39]: This might not be a "publication"

**Pull request merged**